src/panels/config/automation/trigger/types/ha-automation-trigger-time.ts

```diff
@@ -98,6 +106,19 @@ export class HaTimeTrigger extends LitElement implements TriggerElement {
     `;
   }
 
+  protected firstUpdated() {
+    const allEntities = Object.keys(this.hass.states);
+    const input_datetimes = allEntities.filter(
+      (eid) => computeDomain(eid) === "input_datetime"
+    );
+    const sensors = allEntities
+      .filter((eid) => computeDomain(eid) === "sensor")
+      .filter(
+        (eid) => this.hass.states[eid].attributes?.device_class === "timestamp"
+      );
+    this._entities = [...input_datetimes, ...sensors];
```

Comment:
```suggestion
    const allEntities = Object.keys(this.hass.states);
    this._entities = allEntities.filter(
      (eid) =>
        computeDomain(eid) === "input_datetime" ||
        (computeDomain(eid) === "sensor" &&
          this.hass.states[eid].attributes?.device_class === "timestamp")
    );
```

It can be done in only one filter.